genomeGraph/genomeGraph.go

```diff
@@ -48,45 +49,61 @@ type Edge struct {
 func Read(filename string) *GenomeGraph {
 	simpleReader := fileio.NewByteReader(filename)
 	genome := EmptyGraph()
-	var currNode, homeNode, destNode *Node
 	var weight float32
 	var line string
 	var words []string = make([]string, 0, 2)
-	var seqIdx, homeNodeIdx, destNodeIdx uint32
+	var nodeId, homeNodeIdx, destNodeIdx uint32
+	var homeNode, destNode *Node
 	var i int
 
 	for reader, done := fileio.ReadLine(simpleReader); !done; reader, done = fileio.ReadLine(simpleReader) {
 		line = reader.String()
 		switch true {
 		case strings.HasPrefix(line, ">"):
-			seqIdx = common.StringToUint32(line[1:])
-			currNode = &Node{Id: seqIdx}
-			AddNode(genome, currNode)
+			nodeId = common.StringToUint32(line[1:])
+			AddNode(genome, &Node{Id: nodeId})
```

Comment:
This does not need to pass in a pointer, but it allows the easier syntax of:
a = AddNode(genome, a)
similar to:
a = append(a, 7) // when dealing with slices
The idea is that AddNode now copies the values from the node given to it into the slice of Nodes in the genome graph struct.  It then gives the programmer back the memory location where those values were copied to, because that is the copy of the Node that should be used from now on.